posthog/hogql_queries/ai/traces_query_runner.py

```diff
@@ -189,65 +189,82 @@ def _map_person(self, person: tuple[UUID, UUID, datetime, str]) -> LLMTracePerso
-                SELECT
-                    properties.$ai_trace_id as id,
-                    argMin(properties.$ai_input_state, timestamp) as input_state,
-                    argMin(properties.$ai_output_state, timestamp) as output_state,
-                    argMin(ifNull(properties.$ai_span_name, properties.$ai_trace_name), timestamp) as trace_name,
-                    {filter_conditions}
-                FROM events
-                WHERE event = '$ai_trace'
-                    AND properties.$ai_trace_id IS NOT NULL
-                    AND {common_conditions}
-                GROUP BY id
-            ) AS traces
-            ON traces.id = generations.id
+                properties.$ai_trace_id AS id,
+                min(timestamp) AS first_timestamp,
+                tuple(
+                    argMin(person.id, timestamp),
+                    argMin(distinct_id, timestamp),
+                    argMin(person.created_at, timestamp),
+                    argMin(person.properties, timestamp)
+                ) AS first_person,
+                round(
+                    sumIf(toFloat(properties.$ai_latency),
+                          properties.$ai_parent_id IS NULL
+                          OR properties.$ai_parent_id = properties.$ai_trace_id
+                    ), 2
+                ) AS total_latency,
+                sumIf(toFloat(properties.$ai_input_tokens),
+                      event = '$ai_generation'
+                ) AS input_tokens,
+                sumIf(toFloat(properties.$ai_output_tokens),
+                      event = '$ai_generation'
+                ) AS output_tokens,
+                round(
+                    sumIf(toFloat(properties.$ai_input_cost_usd),
+                          event = '$ai_generation'
+                    ), 4
+                ) AS input_cost,
+                round(
+                    sumIf(toFloat(properties.$ai_output_cost_usd),
+                          event = '$ai_generation'
+                    ), 4
+                ) AS output_cost,
+                round(
+                    sumIf(toFloat(properties.$ai_total_cost_usd),
+                          event = '$ai_generation'
+                    ), 4
+                ) AS total_cost,
+                IF({return_full_trace},
+                    arraySort(
+                        x -> x.3,
+                        groupArrayIf(
+                            tuple(uuid, event, timestamp, properties),
+                            event != '$ai_trace'
+                        )
+                    ),
+                    arrayFilter(
+                        x -> x.2 IN ('$ai_metric','$ai_feedback'),
+                        arraySort(x -> x.3,
+                            groupArrayIf(
+                                tuple(uuid, event, timestamp, properties),
+                                event != '$ai_trace'
+                            )
+                        )
+                    )
+                ) AS events,
```

Comment:
Readability nit: equivalent, but streamlined
```suggestion
                arraySort(x -> x.3,
                    groupArrayIf(
                        tuple(uuid, event, timestamp, properties),
                        event IN (if(
                            {return_full_trace},
                            ('$ai_generation','$ai_metric','$ai_feedback'),
                            ('$ai_metric','$ai_feedback')
                        ))
                    )
                ) AS events,
```